bootstrap/src/basis/ioring.c

```diff
@@ -265,86 +297,82 @@ hm_cqring_setup(void *vm, struct io_cqring_offsets const *offsets, hm_cqring_t *
     cqring->cqes = vm + offsets->cqes;
 }
 
-size_t
+static size_t
 hm_ioring_get_vm_size(hm_ioring_t *ioring) {
     return ioring->params.sq_off.array + ioring->params.sq_entries * sizeof(uint32_t);
 }
 
-size_t
+static size_t
 hm_ioring_get_sqes_size(hm_ioring_t *ioring) {
     return ioring->params.sq_entries * sizeof(struct io_uring_sqe);
 }
 
-hm_opt_error_t
+void
 hm_ioring_setup(hm_ioring_t *ioring) {
     hm_opt_error_t oe = HM_OE_NONE;
 
     memset(ioring, 0, sizeof(*ioring));
-    HM_OE_ERRNO_RESULT(oe, ioring->fd, io_uring_setup(32, &ioring->params));
+    ioring->fd = io_uring_setup(HM_IORING_ENTRIES, &ioring->params);
+    if (ioring->fd == -1) {
+        abort();
```

Comment:
We're going to have to recover from these when we integrate executors. I'd imagined this function returning an error, with the current implementation aborting in the caller. That makes later integration easier.

Reply:
I can move the abort into `file.c` or `file.ml`. I think `file.ml` would be the more appropriate of the two.